scripts/lint/spec_lint.py

```
@@ -0,0 +1,74 @@
+        items = sorted(v.items(), key=lambda kv: kv[0])
+        return '{' + ','.join([json.dumps(k, ensure_ascii=False)+":"+canonical(x) for k,x in items]) + '}'
+    if isinstance(v, list):
+        return '[' + ','.join([canonical(x) for x in v]) + ']'
+    if v is None:
+        return 'null'
+    if isinstance(v, bool):
+        return 'true' if v else 'false'
+    if isinstance(v, int):
+        return str(v)
+    if isinstance(v, float):
+        raise SystemExit("ERROR: floats are forbidden in canonical positions; encode as string")
```

Ruff (0.14.3)

Avoid specifying long messages outside the exception class

(TRY003)